### PR TITLE
DHCP4: Allow to emit and receive LPR server

### PR DESCRIPTION
**Pull request by ssahani**
Now option 9 and option 4 is being ignored.
```
pr 01 16:39:21 photon-3 systemd-networkd[476]: Ignoring option DHCP option 9 while parsing.
Apr 01 16:39:21 photon-3 systemd-networkd[476]: DHCP CLIENT (0x340922dd): OFFER
Apr 01 16:39:21 photon-3 systemd-networkd[476]: DHCP CLIENT (0x340922dd): REQUEST (requesting)
Apr 01 16:39:22 photon-3 systemd-networkd[476]: Ignoring option DHCP option 4 while parsing.
Apr 01 16:39:22 photon-3 systemd-networkd[476]: Ignoring option DHCP option 9 while parsing.
Apr 01 16:39:22 photon-3 systemd-networkd[476]: DHCP CLIENT (0x340922dd): ACK
Apr 01 16:39:22 photon-3 systemd-networkd[476]: DHCP CLIENT (0x340922dd): lease expires in 59min 57s
Apr 01 16:39:22 photon-3 systemd-networkd[476]: DHCP CLIENT (0x340922dd): T2 expires in 52min 27s
Apr 01 16:39:22 photon-3 systemd-networkd[476]: DHCP CLIENT (0x340922dd): T1 expires in 29min 59s
```

### CODE REVIEW

File: src/libsystemd-network/sd-dhcp-lease.c [hunk 1]
```diff
@@ -1234,6 +1253,14 @@ int dhcp_lease_load(sd_dhcp_lease **ret, const char *lease_file) {
                         lease->smtp_server_size = r;
         }
 
+        if (lpr_servers) {
+                r = deserialize_in_addrs(&lease->lpr_servers, lpr_servers);
+                if (r < 0)
+                        log_debug_errno(r, "Failed to deserialize log servers %s, ignoring: %m", lpr_servers);
```
Comment (keszybz) [inline src/libsystemd-network/sd-dhcp-lease.c]: C&P here.

File: man/systemd.network.xml [hunk 2]
```diff
@@ -1904,6 +1904,15 @@
         <varname>DNS=</varname>.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>LPRServers=</varname></term>
+        <listitem><para>Specifies a list of Line Printer Daemon Protocol (LPR) servers available
+        to the client.The servers should be listed in order of preference. When configured,
```
Comment (keszybz) [inline man/systemd.network.xml]: Space missing before "The".

File: src/network/networkd-dhcp-server.c [hunk 3]
```diff
@@ -409,6 +458,20 @@ int dhcp4_server_configure(Link *link) {
         if (r < 0)
                 log_link_warning_errno(link, r, "Failed to SMTP server for DHCP server, ignoring: %m");
 
+        if (link->network->n_dhcp_server_log > 0)
+                r = sd_dhcp_server_set_lpr_servers(link->dhcp_server, link->network->dhcp_server_log, link->network->n_dhcp_server_log);
```
Comment (keszybz) [inline src/network/networkd-dhcp-server.c]: Various c&p errors here.

File: src/network/networkd-network.h [hunk 4]
```diff
@@ -153,6 +153,9 @@ struct Network {
         struct in_addr *dhcp_server_smtp;
         unsigned n_dhcp_server_smtp;
 
+        struct in_addr *dhcp_server_log;
+        unsigned n_dhcp_server_log;
```
Comment (keszybz) [inline src/network/networkd-network.h]: ???

**Comment by ssahani** [comment]
> Hmm, it's the four or fifth PR in this series adding new attributes, and for each PR I review, the same typical set of errors is found: copy-and-paste errors, formatting mistakes in the man page. Please, do a careful scan of the diff text before submitting the PR. Otherwise, we're wasting both your time and every reviewer's.
> 
> Note: I didn't read all the way to the end. Might be more issues there.

Ahh sorry. Since they are similar I just tend to miss it. Should be more careful. thanks

**Comment by poettering** [comment]
hmpf. much like for that logging server field: LPR isn't really a thing anymore, is it? it's a protocol without future, everything is IPP now. I doubt any new devices will start deploying LPR info via DHCP these days if they haven't already.

I mean, if there was a speci for encoding an IPP server like this, great, but LPR? not convinced that this should go in.

We shouldn#t support every single option of all specs. Let's focus on the ones people ask for or the ones that have a clear future and purpose. I don't think the LPR stuff qualifies, sorry.

**Pull request closed without merging**